mat/format.go

```diff
@@ -27,6 +28,8 @@ type formatter struct {
 	margin  int
 	dot     byte
 	squeeze bool
+
+	format func(m Matrix, prefix string, margin int, dot byte, squeeze bool, fs fmt.State, c rune)
```

Comment:
why don't we initialize that field with `format` in the `Formatted` function above?
this would somewhat simplify the logic of the `formatter.Format(...)` method down below.

Reply:
The behaviour of default formatting and the foreign formatting is different with `%#v`. This depends on the `nil` state of the `format` field.